content/data-sharing/mutable-and-sendable.md

```diff
@@ -0,0 +1,77 @@
+---
+title: "Mutable and Sendable"
+section: "Data Sharing"
+menu:
+  toc:
+    parent: "data-sharing"
+    weight: 40
+---
+
+## Problem
+
+Suppose you have a central Supervisor actor that manages a very large data structure that contains the work produced by a collection of Worker actors.  However, the Superviser must also respond to data requests (perhaps from the Workers themselves, or in this case, from another set of Requester actors).  As a result, the Superviser's data structure must be both mutable (to update it in response to feedback from Workers) and sendable (to respond to requests from Requesters).  `iso` is the only refcap that accommodates that, but our Supervisor can't use `iso` here because it must keep a reference to the data structure to continue to respond to incoming updates and requests.  How to proceed?
+
+## Solution
+
+```pony
+use mut = "collections"
+use "persistent/collections"
+
+type Datum is List[I32]
+
+
+actor Worker
+  let _supervisor: Supervisor
+
+  new create(supervisor: Supervisor) =>
+  _supervisor = supervisor
+  
+  be do_work(id: USize) =>
+	let result = [as I32: 1; 2; 3; 4] // initial state array
+	// do heavy lifting here; mutate array as needed
+	_supervisor.update(id, Lists[I32](result))
+	
+	
+actor Requester
+  let _supervisor: Supervisor
+  
+  new create(supervisor: Supervisor) =>
+    _supervisor = Supervisor
+    
+  be request(id:USize) =>
+    _supervisor.get(id, this)
+
+  be receive(id: USize, result: Datum) =>
+    // this behavior is called by Supervisor in response to a request
+    // use the result here
+    
+  
+actor Supervisor
+  let _data: mut.Map[USize, Datum]
+  let _default: Datum = Lists[I32]([0; 0; 0; 0])
+  
+  new create() =>
+    _data = mut.Map[USize, Datum](268_435_456) // == 2 ** 28; point being a non-trivially sized data structure, use your imagination
```

Comment:
can you move the comments above the line so it works better on smaller screens?

same below as well